metricbeat/module/elasticsearch/index_summary/data.go

```diff
@@ -47,6 +47,46 @@ var (
 					"bytes": c.Int("memory_in_bytes"),
 				},
 			}),
+			"indexing": c.Dict("indexing", s.Schema{
+				"index": s.Object{
+					"count": c.Int("index_total"),
+					"time": s.Object{
+						"ms": c.Int("index_time_in_millis"),
+					},
+				},
+				// following field is not included in the Stack Monitoring UI mapping
+				"is_throttled": c.Bool("is_throttled"),
```

Comment:
If the field is not needed by the Stack Monitoring UI (@chrisronline may want to confirm) **and** it wasn't a field we were already collecting in `data.go`, then it's safe to not collect it at all. So I would remove this line and others like it.

Reply:
This is correct. We need to not only collect the mappings, but also continue to collect and index any other fields we used to because we often read values from the `source` document that weren't considered when the finalized mappings were produced.

Reply:
@chrisronline So just to clarify, you're saying we **should** collect this field, `is_throttled`?

Reply:
@chrisronline So just to clarify, you're saying we **should** collect this field, `is_throttled` but we just don't need to include it in the `fields.yml` so it doesn't show up in the mappings.

Reply:
Correct

Reply:
Understood :slightly_smiling_face:

Reply:
We don't really even want to index fields that aren't used and we could take another pass on which fields we read from source directly, but we're also currently in talks about potentially indexing as much data as possible (even if the stack monitoring UI doesn't use it) to enable users to build custom dashboards. Nothing is decided but I'd rather leave the functionality as is in case we do go down that path

Reply:
Wait a sec :smile: as far as I understood, we should map this fields in elasticsearch `fields.yml` mapping. But should we still map it in the metricset `fields.yml` file? As a Metricbeat convention, it must be mapped, AFAIK 100% of Metricbeat fields are map in `fields.yml` of their respective metricsets. Metricbeat is not going to fail though, we however have a commonly used python test (like this one in MySQL module https://github.com/elastic/beats/blob/master/metricbeat/module/mysql/test_mysql.py#L47) to check if all fields are map in `fields.yml` but fortunately it's not included in the python test code for the elasticsearch module.

Reply:
`.monitoring-*` indices have dynamic mapping set to `false`, but perhaps `metricbeat-*` handles this differently? In that case, maybe we should add every field we are indexing to `fields.yml` but if the field isn't contained in the used mappings list, we set the mapping to `enabled: false`?

Reply:
@sayden I can try to help clarify about when and where to map fields for stack monitoring metricsets. 😅 

Consider a field, say `elasticsearch.<foo>.bar.baz` that is being collected by the `elasticsearch/<foo>` metricset in `data.go`. For this field:

1. First, you have to decide whether this field just needs to be _stored_ in the `metricbeat-*` indices or whether it needs to be _stored and indexed_ in the `metricbeat-*` indices. If it is the former — just _stored_ —, then you are done with this field. It just needs to be collected in `data.go`. There is no need to add it to any `fields.yml` files. Skip the rest of the steps below.

2. If you are on this step, it means you have decided that the field needs to be _stored and indexed_ in the `metricbeat-*` indices. In this case, go ahead and define it in the `module/elasticsearch/<foo>/_meta/fields.yml` file.

3. Now, you must decide if the field has a corresponding field in the `.monitoring-es-*` index mappings. If it does, let's say that field is `some_thing.some_thing_else.qux.boom`. Go ahead and define this field **as an alias** to `elasticsearch.<foo>.bar.baz` in the `module/elasticsearch/_meta/fields.yml` file.

Hopefully this now clear as mud 😃.

Reply:
Good idea, @chrisronline re: `enabled: false`. I will edit my comment above to reflect this step.